finetune/lora.py

```diff
@@ -119,28 +126,29 @@ def main(fabric: L.Fabric, data_dir: Path, checkpoint_dir: Path, out_dir: Path,
     train_data = torch.load(data_dir / "train.pt")
     val_data = torch.load(data_dir / "test.pt")
 
-    if not any((lora_query, lora_key, lora_value, lora_projection, lora_mlp, lora_head)):
+    if not any((cfg['lora_query'], cfg['lora_key'], cfg['lora_value'], 
+                cfg['lora_projection'], cfg['lora_mlp'], cfg['lora_head'])):
         fabric.print("Warning: all LoRA layers are disabled!")
     config = Config.from_name(
         name=checkpoint_dir.name,
-        r=lora_r,
-        alpha=lora_alpha,
-        dropout=lora_dropout,
-        to_query=lora_query,
-        to_key=lora_key,
-        to_value=lora_value,
-        to_projection=lora_projection,
-        to_mlp=lora_mlp,
-        to_head=lora_head,
+        r=cfg['lora_r'],
+        alpha=cfg['lora_alpha'],
+        dropout=cfg['lora_dropout'],
+        to_query=cfg['lora_query'],
+        to_key=cfg['lora_key'],
+        to_value=cfg['lora_value'],
+        to_projection=cfg['lora_projection'],
+        to_mlp=cfg['lora_mlp'],
+        to_head=cfg['lora_head'],
     )
     checkpoint_path = checkpoint_dir / "lit_model.pth"
     fabric.print(f"Loading model {str(checkpoint_path)!r} with {config.__dict__}")
 
     # group together model config and hyperparams, then log everything to wandb
-    hparams.update(config.__dict__)
-    fabric.logger.log_hyperparams(hparams)
+    cfg.update(config.__dict__)
```

Comment:
It feels a bit circular to instantiate `config` using the entries in `cfg`, and then go back and update `cfg` with the class dict. Is this necessary?

Reply:
My bad for naming the hyperparams as cfg - this is due to the fact that it contains mostly hyperparams but also a few configuration parameters that were not logged before and I wanted to add.
The `config` here is a different configuration (the model config, with additional params), which integrates the hyperparams.
To avoid ambiguities, I will just call `cfg` "hparams" as I did before so it is clear that the two things are non overlapping